akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -778,18 +793,26 @@ private[akka] class ShardRegion(
   }
 
   def receiveCommand(cmd: ShardRegionCommand): Unit = cmd match {
-    case Retry =>
-      sendGracefulShutdownToCoordinator()
 
-      if (shardBuffers.nonEmpty)
-        retryCount += 1
-      if (coordinator.isEmpty)
-        register()
+    case Retry =>
+      // retryCount is used to avoid flooding the logs
+      // it's used inside register() whenever shardBuffers.nonEmpty
+      // therefore we update it if needed on each Retry msg
+      // the reason why it's updated here is because we don't want to increase it on each RegisterRetry, only on Retry
+      if (shardBuffers.nonEmpty) retryCount += 1
```

Comment:
I find this odd. Organical growth? 

I don't want to bring more change to this PR, but I think this should be done inside `register()`. 

I can PR again later. We could define `register` as `def register(retryCount: Int)`. When calling it in the context of `RegisterRetry` we could call `register(0)` to avoid a burst of logging. And when reacting to `Retry` we call it with `register(retryCount += 1)`.

Anyway, for later. Baby steps, small refactorings.